### DISCUSSION TITLE
The owner can't refresh an external library using the API

### DISCUSSION THREAD

**Discussion by ilopmar** [immich-app/immich#10715] [Q&A]
### The bug

I have created an external library and assigned a non-admin user as the owner. I have a script that new photos to that library, so the script triggers a library scan, I don't want to use the scheduled cron job to rescan libraries because I only want to re-scan this specific library.

I send the following request:
```
curl -X POST "https://immich.mydomain.com/api/libraries/${LIBRARY_ID}/scan" \
    -H "Content-Type: application/json" \
    -H "Accept: application/json" \
    -H "x-api-key: ${API_TOKEN}"
```

If the `API_TOKEN` is from the user that owns the library (remember, non-admin user), that API call returns 403. However, if I use a token from an admin user it works.
From the log message this seems to be done on purpose but it's not what I was expecting. If an admin creates a library for a user, that user should be able to "manage" the library.

### The OS that Immich Server is running on

Ubunbu Server 22.04.4

### Version of Immich Server

v1.106.4

### Version of Immich Mobile App

v1.106

### Platform with the issue

- [X] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
Not relevant
```

### Your .env content

```Shell
Not relevant
```

### Reproduction steps

```bash
Explain in the bug
```

### Relevant log output

```shell
immich_server            | [Nest] 17  - 06/30/2024, 10:09:47 AM    WARN [Api:AuthGuard~i0cwj2jn] Denied access to admin only route: /api/libraries/ff330295-dc1a-4f3c-b161-d3f43f73f1d1/scan
immich_server            | [Nest] 17  - 06/30/2024, 10:09:47 AM   DEBUG [Api:HttpExceptionFilter~i0cwj2jn] HttpException(403) {"message":"Forbidden resource","error":"Forbidden","statusCode":403}
```

**Comment by alextran1502** [comment]
This is not a bug since the external library is managed only by the admin user. You don't find the interface for the non-admin user.

**Reply by ilopmar** [reply]
Right, I know there's not option for that in the UI for a non-admin user. I was thinking that maybe for the API that could work.

So, just to make clear, this works as expected and I should use and admin API token for doing the refresh of the external library, right?
